var/spack/repos/builtin/packages/py-setuptools/package.py

```diff
@@ -191,11 +191,16 @@ class PySetuptools(Package, PythonExtension):
     )
 
     extends("python")
+
     depends_on("python@3.7:", when="@59.7:", type=("build", "run"))
     depends_on("python@3.6:", when="@51:", type=("build", "run"))
     depends_on("python@3.5:", when="@45:50", type=("build", "run"))
     depends_on("python@2.7:2.8,3.5:", when="@44", type=("build", "run"))
     depends_on("python@2.7:2.8,3.4:", when="@:43", type=("build", "run"))
+
+    # https://github.com/pypa/setuptools/issues/3661
+    conflicts("python@3.12:", when="@:67")
```

Comment:
Wait should this be `^python`?

Reply:
It could, but it's not necessary anymore. I would make it `^python` and enforce it with an audit (if it is a direct dependency), but not sure if @tgamblin thinks the same.

Reply:
This https://github.com/spack/spack/pull/38447/commits/5ec40aecabd992bd6f7af9a8efe253ab670d8fa9 is where we encoded conflicts so we don't have to enforce `^python`